src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -314,12 +307,39 @@ void testPatchByDatasetUpdate_patchableNoName() {
     when(datasetRegistrationService.patchDataset(any(), any(), any())).thenReturn(dataset);
     when(authUser.getEmail()).thenReturn("test@test.com");
     when(userService.findUserByEmail("test@test.com")).thenReturn(user);
-    when(user.getUserId()).thenReturn(RandomUtils.nextInt(1, 100));
+    when(user.getUserId()).thenReturn(randomInt(1, 100));
+    dataset.setCreateUserId(user.getUserId());
+
+    initResource();
+    try (Response response = resource.patchByDatasetUpdate(authUser, dataset.getDatasetId(),
+        gson.toJson(patch))) {
+      assertEquals(HttpStatusCodes.STATUS_CODE_OK, response.getStatus());
+    }
+  }
+
+  @Test
+  void testPatchByDatasetUpdate_invokeIndexUpdate() throws Exception {
+    Gson gson = GsonUtil.buildGson();
+
+    Dataset dataset = new Dataset();
+    dataset.setDatasetId(randomInt(1, 100));
+    dataset.setName(randomAlphabetic(10));
+    dataset.setIndexedDate(new Date());
+    when(datasetService.findDatasetById(any())).thenReturn(dataset);
+    String newName = randomAlphabetic(20);
+    DatasetPatch patch = new DatasetPatch(newName, List.of());
+
+    when(datasetRegistrationService.patchDataset(any(), any(), any())).thenReturn(dataset);
+    when(authUser.getEmail()).thenReturn("test@test.com");
```

Comment:
This could be avoided by not using a mock for `AuthUser`, since it's a POJO. You could create a test auth user that has its email set for all tests to use.
```
  private final AuthUser authUser = new AuthUser().setEmail("test@test.com");
```